### DISCUSSION TITLE
Remove automatic normalization in Multinomial and Categorical

### DISCUSSION THREAD

**Discussion by ricardoV94** [pymc-devs/pymc3#5246] [General]
This cropped up in #5234 

Should we stop doing automatic normalization of the p parameter? This can hide very wrong inputs such as 
```python
import pymc as pm

with pm.Model() as m:
  x = pm.Multinomial('x', n=5, p=[-2, -2, -2, -2])

m.logp({x: [3, 2, 0, 0]})
# array(-4.62888666)
```

If we decide to keep the automatic normalization, we can at least remove some checks in the logp definition, since they cannot be triggered in this case.

**Comment by ericmjl** [comment]
I'm in favour of this. Explicit is better than implicit!

**Comment by twiecki** [comment]
I think this caused some problems e.g. if you a user specifies `[.3, .3, .3]` where things *almost* line up.

**Comment by fonnesbeck** [comment]
Perhaps have a normalization tolerance, where for example [.33, .33, .33] gets normalized but more egregious values raise an exception? Then at least the user has to be explicit about how close they allow themselves to be.

**Comment by ricardoV94** [comment]
The tolerance can be adjusted with `at.isclose` check, or we can simply do like the Dirichlet value and not even check if the probabilities add up to one (we can just check if they are in the unit range)

**Comment by ricardoV94** [comment]
Here is an idea. Whenever the user creates a Multinomial / Categorical distribution with concrete (numpy) values we check if they are valid and if not we normalize them but also issue a UserWarning along the lines of:

```
p values sum up to 0.9, instead of 1.0. They will be automatically rescaled. You can rescale them directly to get rid of this warning.
```

In the logp or whenever we have symbolic inputs we don't do any invisible normalization, and let it evaluate to -inf as with invalid parameters in other distributions.

I think this covers most of the user cases and does not have big backwards compat issues.

**Reply by twiecki** [reply]
I like that.

**Reply by ericmjl** [reply]
Same here!
